### ISSUE TITLE
Command killed due to excessive memory consumption

### ISSUE THREAD

**Issue by pinggao187** [readthedocs/readthedocs.org#7643]
## Details

* Read the Docs project URL: https://analytics-zoo-doc.readthedocs.io/en/master/
* Build URL (if applicable): https://readthedocs.org/projects/analytics-zoo-doc/builds/12267839/
* Read the Docs username (if applicable): pinggao187

## Expected Result
Project would build without issues.

## Actual Result
/home/docs/checkouts/readthedocs.org/user_builds/analytics-zoo-doc/envs/doc_test/bin/python -m pip install --use-feature 2020-resolver --exists-action=w --no-cache-dir -r doc/requirements-doc.txt

Killed
Command killed due to excessive memory consumption

requirements-doc.txt has not been modified, the build was normal before. can you help me?

**Comment by humitos** [comment]
Looks to be the same issue than #7639. I will try removing the new pip resolver from your project for now.

**Comment by humitos** [comment]
I triggered a new build at https://readthedocs.org/projects/analytics-zoo-doc/builds/12268538/ --let's see if it works.

**Comment by pinggao187** [comment]
sorry, https://readthedocs.org/projects/analytics-zoo-doc/builds/12268538/ This build was triggered after I modified the requirements-doc.txt, a new build https://readthedocs.org/projects/analytics-zoo-doc/builds/12268570/

**Comment by humitos** [comment]
@pinggao187 it seems that you are compiling _a lot_ of dependencies. Besides, it seems that you are installed many dependencies that are not required to build the documentation. I suggest you to mock them all and only install the dependencies that are strictly required to build your documentation. You can take a look at this FAQ, https://docs.readthedocs.io/en/latest/faq.html#i-get-import-errors-on-libraries-that-depend-on-c-modules that explains how to mock the dependencies.

**Comment by pinggao187** [comment]
@humitos thanks for your help! the project build success.

**Comment by humitos** [comment]
Great! I'm closing this issue then.

**Comment by pinggao187** [comment]
I reduced the installation of dependencies and only installed some dependencies. The first time I build it successfully, but I failed to build again. Under the same environment, I built it twice, one succeeded, another failed, and then the build failed many times. How do I solve this issue?

issue:
Killed
Command killed due to excessive memory consumption

https://readthedocs.org/projects/analytics-zoo-doc/builds/12275971/
https://readthedocs.org/projects/analytics-zoo-doc/builds/12277011/
